core/platform-api-client-gwt/che-core-client-gwt-debugger/src/main/java/org/eclipse/che/api/debugger/gwt/client/DebuggerServiceClient.java

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.debugger.gwt.client;
+
+import org.eclipse.che.api.debugger.shared.dto.Breakpoint;
+import org.eclipse.che.api.debugger.shared.dto.DebugSession;
+import org.eclipse.che.api.debugger.shared.dto.Location;
+import org.eclipse.che.api.debugger.shared.dto.StackFrameDump;
+import org.eclipse.che.api.debugger.shared.dto.Value;
+import org.eclipse.che.api.debugger.shared.dto.Variable;
+import org.eclipse.che.api.debugger.shared.dto.action.ResumeAction;
+import org.eclipse.che.api.debugger.shared.dto.action.StartAction;
+import org.eclipse.che.api.debugger.shared.dto.action.StepIntoAction;
+import org.eclipse.che.api.debugger.shared.dto.action.StepOutAction;
+import org.eclipse.che.api.debugger.shared.dto.action.StepOverAction;
+import org.eclipse.che.api.promises.client.Promise;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The client-side service to debug application.
```

Comment:
I'd say it's rather **Client for the service to debug application**.
As for me, unusuall definition - _client-side service_